application-mocca-calendar-api/src/main/java/org/xwiki/contrib/moccacalendar/internal/importJob/ImportJob.java

```diff
@@ -0,0 +1,266 @@
+
+                progressManager.endStep(this);
+                Thread.yield();
+            }
+            batchSave(eventDocuments);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            this.progressManager.popLevelProgress(this);
+        }
+    }
+
+    private void batchSave(List<XWikiDocument> eventDocuments) throws XWikiException
+    {
+        if (!status.isCanceled()) {
+            XWikiContext wikiContext = wikiContextProvider.get();
+            XWiki wiki = wikiContext.getWiki();
+            progressManager.startStep(this);
+            for (XWikiDocument calendarEvent : eventDocuments) {
+                wiki.saveDocument(calendarEvent, wikiContext);
+            }
+            progressManager.endStep(this);
+        }
+    }
+
+    private void createCalendarObjects(XWikiDocument eventDoc, MoccaCalendarEvent component) throws XWikiException
+    {
+        XWikiContext wikiContext = wikiContextProvider.get();
+        DocumentReference eventClassRef =
+            documentReferenceResolver.resolve(EventConstants.MOCCA_CALENDAR_EVENT_CLASS_NAME);
+        BaseObject eventObj = eventDoc.newXObject(eventClassRef, wikiContext);
+
+        addConvertedPropertyToObject(component.getDescription(), eventObj, Syntax.XWIKI_2_1.toIdString(),
+            EventConstants.PROPERTY_DESCRIPTION_NAME);
+        addConvertedPropertyToObject(component.getTitle(), eventObj, Syntax.XWIKI_2_1.toIdString(),
+            EventConstants.PROPERTY_TITLE_NAME);
+
+        int allDay = component.isAllDay() ? 1 : 0;
+        eventObj.set(EventConstants.PROPERTY_ALLDAY_NAME, allDay, wikiContext);
+        if (allDay == 1) {
+            LocalDateTime end = component.getEnd().toInstant().atZone(ZoneOffset.UTC).toLocalDateTime().minusDays(1);
+            eventObj.set(EventConstants.PROPERTY_ENDDATE_NAME, Date.from(end.atZone(ZoneOffset.UTC).toInstant()),
+                wikiContext);
+        } else {
+            eventObj.set(EventConstants.PROPERTY_ENDDATE_NAME, component.getEnd(), wikiContext);
+        }
+        eventObj.set(EventConstants.PROPERTY_STARTDATE_NAME, component.getStart(), wikiContext);
+
+        int recurrenceValue = component.isRecurrent();
+        eventObj.set(EventConstants.PROPERTY_RECURRENT_NAME, recurrenceValue, wikiContext);
+        if (recurrenceValue == 1) {
+            processRecurrence(eventDoc, component);
+        }
+    }
+
+    private void addConvertedPropertyToObject(String content, BaseObject eventObj, String syntax, String property)
+    {
+        String cleanContent = Jsoup.clean(content, Safelist.basic());
+        String convertedContent = htmlConverter.fromHTML(cleanContent, syntax);
+
+        eventObj.set(property, convertedContent, wikiContextProvider.get());
+    }
+
+    private void processRecurrence(XWikiDocument eventDoc, MoccaCalendarEvent component) throws XWikiException
+    {
+        XWikiContext wikiContext = wikiContextProvider.get();
```

Comment:
Could you add a comment about how exactly is this method handling recurrence, what is it doing with the modifications list? Especially since you are doing that conversion